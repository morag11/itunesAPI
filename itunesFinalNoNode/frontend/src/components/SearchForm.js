import React from 'react';
import { useState } from "react"

//this returns the search form to search for media
const SearchForm = (props) => {
    //set states
    const [term, setTerm] = useState("")
    const [media, setMedia] = useState("")
    const [limit, setLimit] = useState(10)

    //gets and sets info from the form
    const searchHandler = (event) => {
        event.preventDefault();
        //ensures everything required is entered
        if(term==="" || media===""){
            alert("Please enter all search terms");
            return
        }
        //these are the parameters required to search
        const params = {
            term: term,
            media: media,
            limit: limit,
        }
        //params are set to state
        props.searchSong(params);
    }

    //media type is selected by radio button, the limit is clicked and the title is entered
    return (
        <div>
            <form onSubmit={searchHandler}>
                <div className="search">
                <label>Search:</label>
                <input className="searchBar" type="text" onChange={(e)=>setTerm(e.target.value)} value={term} ></input>
                </div>
                <div className="mediaTypes">
                <label>Media:</label>
                All <input className="radio" type="radio" name="rdMedia" checked={media==="all"&&"checked"} 
                    value="all" onChange={(e)=>setMedia(e.target.value)}></input>

                Movie <input className="radio" type="radio" name="rdMedia" checked={media==="movie"&&"checked"} 
                    value="movie" onChange={(e)=>setMedia(e.target.value)}></input>

                Music <input className="radio" type="radio" name="rdMedia" checked={media==="music"&&"checked"} 
                    value="music" onChange={(e)=>setMedia(e.target.value)}></input>
                
                TV <input className="radio" type="radio" name="rdMedia" checked={media==="tvShow"&&"checked"} 
                    value="tvShow" onChange={(e)=>setMedia(e.target.value)}></input>
                
                Audiobook <input className="radio" type="radio" name="rdMedia" checked={media==="audiobook"&&"checked"} 
                    value="audiobook" onChange={(e)=>setMedia(e.target.value)}></input>
                
                Software <input className="radio" type="radio" name="rdMedia" checked={media==="software"&&"checked"} 
                    value="software" onChange={(e)=>setMedia(e.target.value)}></input>

                Ebook <input className="radio" type="radio" name="rdMedia" checked={media==="ebook"&&"checked"} 
                    value="ebook" onChange={(e)=>setMedia(e.target.value)}></input>
                </div>
                <label>Limit:</label>
                <input className="limit" type="number" min={1} max={50} onChange={(e)=>setLimit(e.target.value)} value={limit} placeholder="Search..."></input>
                <input className="submit" type="submit" value="Search"></input>
            </form>
        </div>
    )
}

export default SearchForm