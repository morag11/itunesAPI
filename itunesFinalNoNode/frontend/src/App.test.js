// import { render, cleanup } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchForm from './components/SearchForm';
import fetch from 'isomorphic-fetch'
import axios from "axios";

//Snapshot test of initial page
test('Renders without crashing', () => {
    const tree = renderer.create(<SearchForm />).toJSON();
    expect(tree).toMatchSnapshot();
})

//Check the API itself is working
describe('Fetch API Data', () => {
test('Successfully call iTunes Search API', (done) => {
    fetch('https://itunes.apple.com/search?term=beyonce')
    .then(data => {
        expect(data).toBeDefined();
        //if status is between 200 and 300 it was successful
        expect(data.status).toBeGreaterThanOrEqual(200);
        expect(data.status).toBeLessThan(300);
        done();
    });
    });
});

//function mocking API fetch
jest.mock("axios");

export const fetchMedia = async () => {
    try {
        return await axios.get(`https://itunes.apple.com/search?term=beyonce&limit=1`);
    } catch (e) {
        return [];
    }
};

describe("fetchMedia", () => {  
    describe("when API call is successful", () => {
        it("should return media list", async () => {
        // given
            const artistId = 1419227
            axios.get.mockResolvedValueOnce(artistId);
        // when
            const result = await fetchMedia();
            console.log(result);
        // then
            expect(axios.get).toHaveBeenCalledWith(`https://itunes.apple.com/search?term=beyonce&limit=1`);
            expect(result).toEqual(artistId);
        });
    });

    describe("when API call fails", () => {
        it("should return empty list", async () => {
        // given
            const message = "Network Error";
            axios.get.mockRejectedValueOnce(new Error(message));
        // when
            const result = await fetchMedia();
        // then
            expect(axios.get).toHaveBeenCalledWith(`https://itunes.apple.com/search?term=beyonce&limit=1`);
            expect(result).toEqual([]);
        });
    });
});