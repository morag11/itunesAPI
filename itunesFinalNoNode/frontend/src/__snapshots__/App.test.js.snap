// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="search"
    >
      <label>
        Search:
      </label>
      <input
        className="searchBar"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="mediaTypes"
    >
      <label>
        Media:
      </label>
      All 
      <input
        checked={false}
        className="radio"
        name="rdMedia"
        onChange={[Function]}
        type="radio"
        value="all"
      />
      Movie 
      <input
        checked={false}
        className="radio"
        name="rdMedia"
        onChange={[Function]}
        type="radio"
        value="movie"
      />
      Music 
      <input
        checked={false}
        className="radio"
        name="rdMedia"
        onChange={[Function]}
        type="radio"
        value="music"
      />
      TV 
      <input
        checked={false}
        className="radio"
        name="rdMedia"
        onChange={[Function]}
        type="radio"
        value="tvShow"
      />
      Audiobook 
      <input
        checked={false}
        className="radio"
        name="rdMedia"
        onChange={[Function]}
        type="radio"
        value="audiobook"
      />
      Software 
      <input
        checked={false}
        className="radio"
        name="rdMedia"
        onChange={[Function]}
        type="radio"
        value="software"
      />
      Ebook 
      <input
        checked={false}
        className="radio"
        name="rdMedia"
        onChange={[Function]}
        type="radio"
        value="ebook"
      />
    </div>
    <label>
      Limit:
    </label>
    <input
      className="limit"
      max={50}
      min={1}
      onChange={[Function]}
      placeholder="Search..."
      type="number"
      value={10}
    />
    <input
      className="submit"
      type="submit"
      value="Search"
    />
  </form>
</div>
`;
